core/dbt/artifacts/resources/v1/model.py

```diff
@@ -12,15 +12,68 @@
 from dbt.artifacts.resources.v1.config import NodeConfig
 from dbt_common.contracts.config.base import MergeBehavior
 from dbt_common.contracts.constraints import ModelLevelConstraint
+from dbt_common.contracts.util import Mergeable
 from dbt_common.dataclass_schema import ExtensibleDbtClassMixin, dbtClassMixin
 
 
+class ModelFreshnessUpdatesOnOptions(enum.Enum):
+    all = "all"
+    any = "any"
+
+
+@dataclass
+class ModelBuildAfter(ExtensibleDbtClassMixin):
+    updates_on: ModelFreshnessUpdatesOnOptions = ModelFreshnessUpdatesOnOptions.any
+    count: int = 0
+    period: TimePeriod = TimePeriod.hour
+
+
+@dataclass
+class ModelFreshness(ExtensibleDbtClassMixin, Mergeable):
+    build_after: ModelBuildAfter = field(default_factory=ModelBuildAfter)
+
+
+def merge_model_freshness(*thresholds: Optional[ModelFreshness]) -> Optional[ModelFreshness]:
+    if not thresholds:
+        return None
+
+    current_merged_value: Optional[ModelFreshness] = thresholds[0]
+
+    for i in range(1, len(thresholds)):
+        base = current_merged_value
+        update = thresholds[i]
+
+        if base is not None and update is not None:
+            # When both base and update freshness are defined,
+            # create a new ModelFreshness instance using the build_after from the 'update'.
+            # This effectively means 'update's build_after configuration takes precedence.
+            merged_freshness_obj = base.merged(update)
+            if (
+                base.build_after.updates_on == ModelFreshnessUpdatesOnOptions.all
+                or update.build_after.updates_on == ModelFreshnessUpdatesOnOptions.all
+            ):
+                merged_freshness_obj.build_after.updates_on = ModelFreshnessUpdatesOnOptions.all
+            current_merged_value = merged_freshness_obj
+        elif base is None and update is not None:
+            # If the current merged value is None but the new update is defined,
+            # take the update.
+            current_merged_value = update
+        else:
+            # This covers cases where 'update' is None (regardless of 'base'),
+            # or both 'base' and 'update' are None.
+            # The result of the pair-merge is None.
+            current_merged_value = base
+
+    return current_merged_value
+
+
 @dataclass
 class ModelConfig(NodeConfig):
     access: AccessType = field(
         default=AccessType.Protected,
         metadata=MergeBehavior.Clobber.meta(),
     )
+    freshness: Optional[ModelFreshness] = None
```

Comment:
I can't comment on the relevant line, but on 103 there is a top level freshness property for models. This was recently added, never released, and never documented. Can we safely rip it out?

Reply:
we should rip out the top-level-property for freshness for models, and only support it nested under config